test/shopify.test.js

```diff
@@ -367,4 +373,168 @@ describe('Shopify', () => {
+          restoreRate: 50.0
+        }
+      }
+    };
+
+    const scope = nock(`https://${shopName}.myshopify.com`, {
+      reqheaders: {
+        'User-Agent': `${pkg.name}/${pkg.version}`,
+        'X-Shopify-Access-Token': accessToken
+      }
+    });
+
+    afterEach(() => expect(nock.isDone()).to.be.true);
+
+    it('returns a RequestError when the request fails', () => {
+      const message = 'Something wrong happened';
+
+      scope
+        .post(path)
+        .replyWithError(message);
+
+      return shopify.graphql(dummyData).then(() => {
+        throw new Error('Test invalidation');
+      }, err => {
+        expect(err).to.be.an.instanceof(got.RequestError);
+        expect(err.message).to.equal(message);
+      });
+    });
+
+    it('returns a RequestError when timeout expires (1/2)', () => {
+      const shopify = new Shopify({ shopName, accessToken, timeout: 100 });
+
+      shopify.baseUrl.hostname = '192.0.2.1';
+
+      return shopify.graphql(dummyData).then(() => {
+        throw new Error('Test invalidation');
+      }, err => {
+        expect(err).to.be.an.instanceof(got.RequestError);
+        expect(err.message).to.equal('Request timed out');
+      });
+    });
+
+    it('returns a RequestError when timeout expires (2/2)', () => {
+      const shopify = new Shopify({ shopName, accessToken, timeout: 100 });
+
+      scope
+        .post(path)
+        .delayBody(200)
+        .reply(200, {});
+
+      return shopify.graphql(dummyData).then(() => {
+        throw new Error('Test invalidation');
+      }, err => {
+        expect(err).to.be.an.instanceof(got.RequestError);
+        expect(err.message).to.include('Request timed out');
+      });
+    });
+
+    it('returns an HTTPError when the server response code is not 2xx', () => {
+      scope
+        .post(path)
+        .reply(400, {});
+
+      return shopify.graphql(dummyData).then(() => {
+        throw new Error('Test invalidation');
+      }, err => {
+        expect(err).to.be.an.instanceof(got.HTTPError);
+        expect(err.message).to.equal('Response code 400 (Bad Request)');
+      });
+    });
+
+    it('uses basic auth as intended', () => {
+      const shopify = new Shopify({ shopName, apiKey, password });
+
+      nock(`https://${shopName}.myshopify.com`, {
+        reqheaders: {
+          'User-Agent': `${pkg.name}/${pkg.version}`
+        },
+        badheaders: ['X-Shopify-Access-Token']
+      }).post(path)
+        .basicAuth({ user: apiKey, pass: password })
+        .reply(200, {});
+
+      return shopify.graphql(dummyData);
+    });
+
+    it('updates callGraphqlLimits if the extensions attribute exists', () => {
+      scope
+        .post(path)
+        .reply(200, { extensions: dummyExtensions });
+
+      return shopify.graphql(dummyData)
+        .then(() => {
+          expect(shopify.callGraphqlLimits).to.deep.equal({
+            remaining: 997,
+            current: 3,
+            max: 1000.0
```

Comment:
Can this be a non integer?

Reply:
Its weird, but this is an example of a response I got from Shopify:
```
{"data":{"customers":{"edges":[{"node":{"displayName":"John Doe","totalSpent":"8800.00"}}]}},"extensions":{"cost":{"requestedQueryCost":7,"actualQueryCost":3,"throttleStatus":{"maximumAvailable":1000.0,"currentlyAvailable":997,"restoreRate":50.0}}}}
```

So, seems it can be a float